### ISSUE TITLE
Fluentd dies when executing example

### ISSUE THREAD

**Issue by pmoncadaisla** [kazegusuri/fluent-plugin-prometheus#2]
When executing

`bundle exec fluentd -c misc/fluentd_sample.conf -v`

it dies:

```
2016-03-29 15:09:14 -0400 [info]: fluent/supervisor.rb:465:read_config: reading config file path="misc/fluentd_sample.conf"
2016-03-29 15:09:14 -0400 [info]: fluent/supervisor.rb:331:supervise: starting fluentd-0.12.22
2016-03-29 15:09:14 -0400 [info]: fluent/engine.rb:126:block in configure: gem 'fluentd' version '0.12.22'
2016-03-29 15:09:14 -0400 [info]: fluent/engine.rb:126:block in configure: gem 'fluent-plugin-prometheus' version '0.1.1'
2016-03-29 15:09:14 -0400 [info]: fluent/agent.rb:140:add_filter: adding filter pattern="nginx" type="prometheus"
2016-03-29 15:09:14 -0400 [info]: fluent/agent.rb:128:add_match: adding match pattern="nginx" type="copy"
2016-03-29 15:09:14 -0400 [debug]: plugin/out_copy.rb:44:block in configure: adding store type="stdout"
2016-03-29 15:09:14 -0400 [info]: fluent/agent.rb:140:add_filter: adding filter pattern="nginx_proxy" type="prometheus"
2016-03-29 15:09:14 -0400 [info]: fluent/agent.rb:128:add_match: adding match pattern="nginx_proxy" type="copy"
2016-03-29 15:09:14 -0400 [debug]: plugin/out_copy.rb:44:block in configure: adding store type="stdout"
2016-03-29 15:09:14 -0400 [info]: fluent/root_agent.rb:147:add_source: adding source type="prometheus"
2016-03-29 15:09:14 -0400 [info]: fluent/root_agent.rb:147:add_source: adding source type="prometheus_monitor"
2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:372:rescue in main_process: unexpected error error="wrong number of arguments (3 for 1)"
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/plugin/filter_record_transformer.rb:195:in `prepare_placeholders'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/lib/fluent/plugin/in_prometheus_monitor.rb:20:in `configure'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/root_agent.rb:154:in `add_source'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/root_agent.rb:95:in `block in configure'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/root_agent.rb:92:in `each'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/root_agent.rb:92:in `configure'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/engine.rb:129:in `configure'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/engine.rb:103:in `run_configure'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/supervisor.rb:483:in `run_configure'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/supervisor.rb:154:in `block in start'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/supervisor.rb:360:in `call'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/supervisor.rb:360:in `main_process'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/supervisor.rb:333:in `block in supervise'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/supervisor.rb:332:in `fork'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/supervisor.rb:332:in `supervise'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/supervisor.rb:150:in `start'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/lib/fluent/command/fluentd.rb:173:in `<top (required)>'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/bin/fluentd:5:in `require'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/gems/fluentd-0.12.22/bin/fluentd:5:in `<top (required)>'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/bin/fluentd:23:in `load'
  2016-03-29 15:09:14 -0400 [error]: fluent/supervisor.rb:332:fork: /root/fluent-plugin-prometheus/vendor/bundle/ruby/1.9.1/bin/fluentd:23:in `<main>'
2016-03-29 15:09:14 -0400 [info]: fluent/supervisor.rb:348:supervise: process finished code=256
2016-03-29 15:09:14 -0400 [warn]: fluent/supervisor.rb:351:supervise: process died within 1 second. exit.

```

anything I might be doing wrong?

**Comment by wyukawa** [comment]
I am having the same issue.

Because fluent-plugin-prometheus uses ```prepare_placeholders``` of fluentd v0.14.0.pre.1, it doesn't work on fluentd 0.12.

https://github.com/kazegusuri/fluent-plugin-prometheus/blob/master/lib/fluent/plugin/prometheus.rb#L76
https://github.com/fluent/fluentd/blob/v0.14.0.pre.1/lib/fluent/plugin/filter_record_transformer.rb#L172
https://github.com/fluent/fluentd/blob/v0.12.22/lib/fluent/plugin/filter_record_transformer.rb#L195

Maybe I think that fluent-plugin-prometheus should not use ```prepare_placeholders``` of fluentd v0.14.0.pre.1 because ```prepare_placeholders``` is not public API.

**Comment by kazegusuri** [comment]
Sorry for late reply. I will check this later (soone).

2016-04-04 23:16 GMT+09:00 wyukawa <notifications@github.com>:

**Comment by kazegusuri** [comment]
I have released v0.1.2 which fixed this issue. It depends the latest version of fluentd (>= v0.12.21) or fluent-plugin-record-reformer for fluentd v0.10.




> Maybe I think that fluent-plugin-prometheus should not use prepare_placeholders of fluentd v0.14.0.pre.1 because prepare_placeholders is not public API.

I know the API is not public and the API sometimes changes, which break this plugin (actually this is second time). But expanding values of record is a little complex and I want to provide same interface to handle record. Some plugins have built-in expander or modifier interface, which is confusing end-users.

**Comment by wyukawa** [comment]
I confirm that the problem is resolved. Thanks!
